persistent-test/src/CompositeTest.hs

```diff
@@ -12,7 +12,11 @@ import Init
 
 
 -- mpsGeneric = False is due to a bug or at least lack of a feature in mkKeyTypeDec TH.hs
-share [mkPersist persistSettings { mpsGeneric = False }, mkMigrate "compositeMigrate", mkDeleteCascade persistSettings { mpsGeneric = False }] [persistLowerCase|
+share
+    [ mkPersist persistSettings { mpsGeneric = False }
+    , mkMigrate "compositeMigrate"
+    , mkDeleteCascade persistSettings { mpsGeneric = False }
```

Comment:
Currently this fails with:

```

    /home/matt/Projects/persistent/persistent-test/src/CompositeTest.hs:15:1: error:
        • Couldn't match type ‘Key TestParent’ with ‘[Char]’
            arising from a use of ‘TestChildName’
        • In the first argument of ‘Filter’, namely ‘TestChildName’
          In the expression:
            ((Filter TestChildName) (FilterValue key_aGp3)) Eq
          In the first argument of ‘deleteCascadeWhere’, namely
            ‘[((Filter TestChildName) (FilterValue key_aGp3)) Eq,
              ((Filter TestChildName2) (FilterValue key_aGp3)) Eq,
              ((Filter TestChildAge) (FilterValue key_aGp3)) Eq]’
       |
    15 | share
       | ^^^^^...
```

Previously, we weren't generating *anything* here. But now we're generating something like:

```
deleteWhere [TestChildName ==. k, TestChildName2 ==. k, TestChildAge ==. k]
```

which is obviously not right. So the "fix" in thsi PR is borked.